packages/qdrant-loader/src/qdrant_loader/core/file_conversion/clean_xlsx_converter.py

```
@@ -0,0 +1,164 @@
+def _is_blank_dataframe(df: pd.DataFrame) -> bool:
+    """Return True for a DataFrame that holds no real data after cleaning.
+
+    Used on per-subtable frames, not full sheets — the name avoids the older
+    "_should_skip_sheet" misnomer.
+    """
+    if df.empty:
+        return True
+    return bool(df.isna().all().all())
+
+
+def _cell_to_str(value: Any) -> str:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)